### ISSUE TITLE
Error condition: Replacement forces bad HTML

### ISSUE THREAD

**Issue by jdickey** [TheProlog/prolog-services-replace_content#5]
This wasn't as clearly differentiated from "invalid endpoints" in the original README as it could have been.

What if the endpoints are validated (by sticking HTML comments in at the endpoints and making sure the result is still parseable), but after replacing the content, the "final" result is invalid?

For a very contrived example:

``` ruby
source_html = '<ul><li>First item</li><li>Second item</li></ul>'
begin_ep = source_html.index 'Second'
end_ep = source_html.index '</ul>'
endpoints = (begin_ep...end_ep)
converter = Prolog::Services::ReplaceContent.new(content: source_html,
                                                 endpoints: endpoints,
                                                 replacement: 'basic' )
```

The resulting markup would (try to) be

``` html
<ul><li>First item</li><li>basic</ul>
```

Most modern Web browsers' parsers, accustomed to often ridiculously-malformed tag soup, would figure out that "hey, there really should be an `</li>` after `basic` there", but the one exception is most likely to reveal itself while doing your most important demo. Much better to make the tools we're using happy (e.g., Ox or anything other than Nokogiri) and reject anything that breaks form or validity. (Now you see why Nokogiri is so big and slow. Like the tap-dancing bear, the wonder is that it works as well as it does.)

This setup would fail to trigger the error conditions we've tested thus far (invalid initial content, endpoints in unfortunate places), so we need to add code to address it. Thus, this issue and the resulting PR.

**Comment by jdickey** [comment]
There was a long, rambling comment here, that has been moved out to [this page](https://github.com/TheProlog/prolog-services-replace_content/wiki/Code-and-Process-Smells) on the project Wiki (and not yet re-edited properly, as we write this).

This was done as we chose to refocus development of this Gem, which will now have at least two releases:
- an initial 0.1.0 release, which may well be bumped to 1.0.0 once included in production code; and
- an indefinite-future release (which our conventions refer to as `The Glorious Future™`), which includes features we _want_ to ship but simply don't presently have the resources to accomplish.

The use of Markdown for either source content or replacement content has now been punted to TGF. We are henceforth assuming that we are dealing _exclusively_ with valid, well-formed HTML.
